src/ConditionsResolver/Parser/ConditionalParameterParser.cs

```diff
@@ -243,29 +244,79 @@ private void ClearWorkflowParser()
         private (string? Result, ParameterContext Context) ResolveExecutionTasks(string value)
         {
             var subValue = value.Trim().Substring(ExecutionsTask.Length, value.Length - ExecutionsTask.Length);
-            var subValues = subValue.Split('[', ']');
+            var subValues = subValue.Split('.');
             var id = subValues[1].Trim('\'');
 
-            var task = WorkflowInstance?.Tasks.First(t => t.TaskId == id);
+            var task = WorkflowInstance?.Tasks.FirstOrDefault(t => t.TaskId == id);
 
-            if (task is null || task is not null && !task.Metadata.Any())
+            if (task is null)
             {
                 return (Result: null, Context: ParameterContext.TaskExecutions);
             }
 
-            var metadataKey = subValues[2].Split('\'')[1];
+            var subValueKey = subValues[2];
+            string? keyValue = null;
 
-            if (task is not null && task.Metadata.ContainsKey(metadataKey))
+            if (subValues.Length > 3)
             {
-                var result = task.Metadata[metadataKey];
+                keyValue = subValues[3]?.Split('\'')[1];
+            }
+
+            var resultStr = null as string;
+            switch (subValueKey.ToLower())
+            {
+                case "task_id":
```

Comment:
maybe we should implement a constants class for all these constant strings ?
could include patient details strings below?

Reply:
Good idea, pulled them out to a constants file